src/components/scenes/Fio/FioAddressRegisterSelectWalletScene.tsx

```diff
@@ -151,12 +148,16 @@ export class FioAddressRegisterSelectWallet extends React.Component<Props, Local
         this.props.onSelectWallet(walletId, paymentCurrencyCode)
 
         const wallet = account.currencyWallets[walletId]
-        const exchangeDenomination = getExchangeDenomByCurrencyCode(wallet.currencyConfig, paymentCurrencyCode)
-        const exchangeAmount = allPaymentInfo[paymentCurrencyCode].amount
-        let nativeAmount = mul(exchangeAmount, exchangeDenomination.multiplier)
-        nativeAmount = toFixed(nativeAmount, 0, 0)
+        const { amount: exchangeAmount, address: paymentAddress } = allPaymentInfo[paymentCurrencyCode]
+
+        const cryptoAmount = new CryptoAmount({
+          exchangeAmount,
+          currencyCode: paymentCurrencyCode,
+          currencyConfig: wallet.currencyConfig
+        })
+
+        const { tokenId, nativeAmount } = cryptoAmount
```

Comment:
Oooh, nice. Your new class may be even more useful than I expected. I like where this is going.